src/WaveTrack.cpp

```diff
@@ -1667,7 +1667,7 @@ bool WaveTrack::HandleXMLTag(const wxChar *tag, const wxChar **attrs)
          }
          else if (!wxStrcmp(attr, wxT("linked")) &&
                   XMLValueChecker::IsGoodInt(strValue) && strValue.ToLong(&nValue))
-            SetLinked(nValue != 0);
+            SetLinkType(static_cast<Track::LinkType>(nValue));
```

Comment:
Will the LinkType enumeration need more values in future versions of Audacity?  Maybe.

So consider, what if we are loading a future file, and there is a value not within the existing LinkType enumeration of Audacity 3.1?

The future value must be corrected to one of the known values, or loading the file should abort.

If aborting isn't acceptable, what's the best guess we can make now about future designs we haven't done yet?  A crazy question maybe, but we have to decide something now.  And I would say:  assume the unknown value can map to "Aligned" which is the expected usual case.

(What if the guess is wrong?  Then Audacity 3.1 ought to have some way to let the user correct the situation, changing incorrectly "Aligned" tracks to "Grouped" with something in the drop-down menu of the Track Control Panel.  Is that in the scope of this issue?  No, so don't act on this suggestion yet.  But we should argue it in the next design review.)

Reply:
I didn't see that enum (or boolean value in the past) beeing used somewhere except WaveTrack actually. And i don't see any sense in linking label tracks for example (maybe we would find use for this in future, but that's a speculation). And note that value is loaded/stored in WaveTrack but belongs to the Track class.

>The future value must be corrected to one of the known values, or loading the file should abort.

That's why i don't like modifying meaning of existing entities :) but make sense to me, and... 'Aligned' is more strict condition compared to 'Grouped', and 'Grouped' is stricter than 'None', and who knows which type of link we would decide to have in future (maybe we'll remove it all), but droping to 'None' is the safest option here i think. Also there is a version check exist for project files

Reply:
> I didn't see that enum (or boolean value in the past) beeing used somewhere except WaveTrack context actually. And i don't see any sense in linking label tracks for example (maybe we would find use for this in future, but that's a speculation). And note that value is loaded/stored in WaveTrack but belongs to the Track class.
> 
> > The future value must be corrected to one of the known values, or loading the file should abort.
> 
> That's why i don't like modifying meaning of existing entities :) but make sense to me, and... 'Aligned' is more strict condition compared to 'Grouped', and 'Grouped' is stricter than 'None', and who knows which type of link we would decide to have in future (maybe we'll remove it all), but droping to 'None' is the safest option here i think. Also there is a version check exist for project files

There is a version check, but Audacity has very rarely made incompatible changes in version, and I don't know yet that we will decide to do so.

Maybe you have the better argument, that if there is an unknonwn future value for this field read into version 3.1 -- then the worst assumptions should be made, and it should be corrected to None not Aligned.  If that is wrong, let the user join mono channels back into stereo.  (Ah, but again -- should they have a choice to make the stereo tracks Aligned not simply Grouped?)

But the larger point remains:  sometihing must be done in 3.1 at load time, to be sure that the value read from the file maps to some valid value of the small enumeration of three values.  That isn't done yet here.

Reply:
One more matter:  you still need the loading procedure to defend against unexpected values for "linked" that are not within the enumeration.

You can map all values outside { 0, 1, 2, 3 } to Grouped (which means, making the least assumptions about Non-mono tracks.)